src/main/java/org/sonarlint/intellij/ui/WhatsInThisViewPanel.kt

```diff
@@ -0,0 +1,107 @@
+ */
+package org.sonarlint.intellij.ui
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.util.ui.JBUI
+import java.awt.CardLayout
+import java.awt.GridBagConstraints
+import java.awt.GridBagLayout
+import javax.swing.JLabel
+import javax.swing.JPanel
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.common.vcs.VcsListener
+import org.sonarlint.intellij.core.ProjectBindingManager
+import org.sonarlint.intellij.ui.UiUtils.Companion.runOnUiThread
+import org.sonarlint.intellij.util.HelpLabelUtils.Companion.createCurrentFileHelp
+import org.sonarlint.intellij.util.HelpLabelUtils.Companion.createCurrentFileHelpNotConnected
+import org.sonarlint.intellij.util.runOnPooledThread
+
+class WhatsInThisViewPanel(project: Project) {
+    private val NOT_CONNECTED = "Not Connected"
+    private val CONNECTED = "Connected"
+    private var project: Project? = null
+
+    var panel: JPanel? = null
+    private var layout: CardLayout? = null
+
+    init {
+        layout = CardLayout()
+        panel = JPanel(layout)
+        this.project = project
+        createPanel()
+        switchCards()
+        CurrentFileStatusPanel.subscribeToEventsThatAffectCurrentFile(project) { this.switchCards() }
+        project.messageBus.connect(project).subscribe<VcsListener>(
+            VcsListener.TOPIC,
+            VcsListener { _: Module?, _: String? ->
+                runOnUiThread(
+                    project
+                ) { this.switchCards() }
+            })
+    }
+
+    private fun createPanel() {
+        val notConnectedCardWrapper = JPanel(GridBagLayout())
+        val connectedCardWrapper = JPanel(GridBagLayout())
+        val helpLabelConnected = createCurrentFileHelp()
+        val helpLabelNotConnected = createCurrentFileHelpNotConnected()
+
+        val gcs = GridBagConstraints(
+            GridBagConstraints.CENTER, 0, 1, 1, 0.0, 0.0, GridBagConstraints.CENTER, GridBagConstraints.NONE,
+            JBUI.insets(2, 5, 2, 0), 0, 0
+        )
+
+        val helpLabelTextConnected = JLabel("What's in this view")
+        val helpLabelTextNotConnected = JLabel("What's in this view")
+
+        notConnectedCardWrapper.add(helpLabelTextNotConnected)
+        notConnectedCardWrapper.add(helpLabelNotConnected, gcs)
+        connectedCardWrapper.add(helpLabelTextConnected)
+        connectedCardWrapper.add(helpLabelConnected, gcs)
+
+        panel!!.add(notConnectedCardWrapper, NOT_CONNECTED)
+        panel!!.add(connectedCardWrapper, CONNECTED)
+        layout!!.show(panel, NOT_CONNECTED)
+    }
+
+    private fun switchCards() {
+        ApplicationManager.getApplication().assertIsDispatchThread()
+
+        // Checking connected mode state may take time, so lets move from EDT to pooled thread
+        runOnPooledThread(project!!) {
```

Comment:
There are several places where you use the `!!` operator, do you need it? `project` is not nullable